src/daft-core/src/series/ops/list.rs

```diff
@@ -175,4 +176,45 @@ impl Series {
             ))),
         }
     }
+
+    /// Given a series of `List` or `FixedSizeList`, return the count of distinct elements in the list.
+    ///
+    /// # Note
+    /// `NULL` values are not counted.
+    ///
+    /// # Example
+    /// ```txt
+    /// [[1, 2, 3], [1, 1, 1], [NULL, NULL, 5]] -> [3, 1, 1]
+    /// ```
+    pub fn list_unique_count(&self) -> DaftResult<Self> {
+        let field = Field::new(self.name(), DataType::UInt64);
+        match self.data_type() {
+            DataType::List(..) => {
+                let iter = self.list()?.into_iter().map(|sub_series| {
+                    let sub_series = sub_series?;
+                    let length = sub_series
+                        .build_probe_table()
+                        .expect("Building the probe table should always work")
```

Comment:
That would require a *complete* materialization of each element in the iterator. Since iterators are lazy (and the actual "driver" of this iterator would be the internals of the `arrow2` crate), if I actually want to eagerly operate on this iterator, I would need to `.collect()` it in this function. If I do that, then I can actually realize the output of each closure, and if any of them are erroneous, exit early.

In other words, we would need a conversion from `Iterator<Item = Result<X>>` to `Result<Iterator<Item = X>>`.

Ping @andrewgazelka and @samster25 on if you guys have any ideas on how to perform an early exit if a closure in a `Map` returns an erroneous value *without* materializing everything.

Or maybe materializing everything is not such a bad thing?

Reply:
Actually, I found a good middle ground for this one. I will add a new `from_fallible_iter` on UInt64Array. The iterator will bail early if it encounters an error while it's driving the iterator.

Reply:
Resolving for now.